core/modules/rproc/InfileMerger.cc

```diff
@@ -319,34 +303,38 @@ bool InfileMerger::merge(std::shared_ptr<proto::WorkerResponse> response) {
             }
         }
     }
-    LOGS(_log, LOG_LVL_INFO, "&&&mergeCount " << mergeCount.mergeCount << " mergeDur=" << mergeDur.count());
+    LOGS(_log, LOG_LVL_INFO, "mergeDur=" << mergeDur.count());
     return ret;
 }
 
-/* &&&INFILE
-bool InfileMerger::_applyMysql(std::string const& query) {
-    std::lock_guard<std::mutex> lock(_mysqlMutex);
-    if (!_mysqlConn.connected()) {
+
+bool InfileMerger::_applyMysqlMyIsam(std::string const& query) {
+    std::unique_lock<std::mutex> lock(_mysqlMutex);
+    for (int j=0;!_mysqlConn.connected(); ++j) {
         // should have connected during construction
         // Try reconnecting--maybe we timed out.
-        if (!_setupConnection()) {
-            LOGS(_log, LOG_LVL_ERROR, "InfileMerger::_applyMysql _setupConnection() failed!!!");
-            return false; // Reconnection failed. This is an error.
+        if (!_setupConnectionMyIsam()) {
+            if (j < 10) {
+                lock.unlock();
+                sleep(1);
```

Comment:
Why all these magic numbers here? Application-level connection timeouts should be configurable.